book/B-embedding-git/sections/go-git.asc

```diff
@@ -0,0 +1,84 @@
+=== go-git
+
+(((go-git)))((("Go")))
+In case you want to integrate Git into a service written in Golang, there also is a pure Go library implmentation. In contrast to using a Libgit2 bindings, this implementation does not have any native dependencies, not prone to memory management errors and is transparent for standard Golang performance analysis tooling like CPU, Memory profilers, race detector, etc.
+
+go-git is focused on extensibility, compatibility and supports most of the plumbing APIs, wich is documented https://github.com/src-d/go-git/blob/master/COMPATIBILITY.md[]
+
+Here is a basic example of using Go APIs
+
+[source, go]
+-----
+import 	"gopkg.in/src-d/go-git.v4"
+
+r, err := git.PlainClone("/tmp/foo", false, &git.CloneOptions{
+    URL:      "https://github.com/src-d/go-git",
+    Progress: os.Stdout,
+})
+-----
+
+As soon as you have a `Repository` instance, 
+
+
+[source, go]
+-----
+// retrieves the branch pointed by HEAD
+ref, err := r.Head()
+
+// get the commit object, pointed by ref
+commit, err := r.CommitObject(ref.Hash())
+
+// retrieves the commit history
+history, err := commit.History()
+
+// iterates over the commits and print each
+for _, c := range history {
+    fmt.Println(c)
+}
+-----
+
+
+==== Advanced Functionality
+
+go-git has few notable advanced features, one of wich is a pluggable storage system, similar to Libgit2 backends, with a default implementation of in-memory storage.
```

Comment:
This is a lot of commas. I'd suggest splitting it up like this:

> go-git has few notable advanced features, one of wich is a pluggable storage system, which is similar to Libgit2 backends. The default implementation is in-memory storage, which is very fast.

Reply:
another typo: "wich"